packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -0,0 +1,1144 @@
+        return builder(context, kAlwaysCompleteAnimation);
+      case NavigationBarDestinationLabelBehavior.alwaysHide:
+        return builder(context, kAlwaysDismissedAnimation);
+      case NavigationBarDestinationLabelBehavior.onlyShowSelected:
+        return _CurvedAnimationBuilder(
+          animation: info.selectedAnimation,
+          curve: Curves.easeInOutCubicEmphasized,
+          reverseCurve: Curves.easeInOutCubicEmphasized.flipped,
+          builder: (BuildContext context, Animation<double> curvedAnimation) {
+            return builder(context, curvedAnimation);
+          },
+        );
+    }
+  }
+}
+
+/// Semantics widget for a navigation bar destination.
+///
+/// Requires a [NavigationBarDestinationInfo] parent (normally provided by the
+/// [NavigationBar] by default).
+///
+/// Provides localized semantic labels to the destination, for example, it will
+/// read "Home, Tab 1 of 3".
+///
+/// Used by [NavigationBarDestinationBuilder].
+class NavigationBarDestinationSemantics extends StatelessWidget {
+  /// Adds the the appropriate semantics for navigation bar destinations to the
+  /// [child].
+  const NavigationBarDestinationSemantics({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  /// The widget that should receive the destination semantics.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final NavigationBarDestinationInfo destinationInfo = NavigationBarDestinationInfo.of(context);
+    // The AnimationStatusBuilder will make sure that the semantics update to
+    // "selected" when the animation status changes.
+    return StatusTransitionWidgetBuilder(
+      animation: destinationInfo.selectedAnimation,
+      builder: (BuildContext context, Widget? child) {
+        return Semantics(
+          selected: destinationInfo.selectedAnimation.isForwardOrCompleted,
+          container: true,
+          child: child,
+        );
+      },
+      child: Stack(
+        alignment: Alignment.center,
+        children: <Widget>[
+          child,
+          Semantics(
+            label: localizations.tabLabel(
+              tabIndex: destinationInfo.destinationNumber + 1,
+              tabCount: destinationInfo.totalNumberOfDestinations,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// Tooltip widget for use in a [NavigationBar].
+///
+/// It appears just above the navigation bar when one of the destinations is
+/// long pressed.
+class NavigationBarDestinationTooltip extends StatelessWidget {
+  /// Adds a tooltip to the [child] widget.
+  const NavigationBarDestinationTooltip({
+    Key? key,
+    required this.message,
+    required this.child,
+  }) : super(key: key);
+
+  /// The text that is rendered in the tooltip when it appears.
+  ///
+  /// If [message] is null, no tooltip will be used.
+  final String? message;
```

Comment:
I don't think this needs to be nullable since you always pass a string in (whether it is `tooltip` or `label`)